### PR TITLE
Remove method from ChannelOutboundInvoker protocol that can be extens…

### PR DESCRIPTION
**Pull request by normanmaurer**
…ions.

Motivation:

We should not have methods in the ChannelOutboundInvoker that can easily implemented as extension and that never need to be overriden.

Modifications:

Remove methods that return EventLoopFuture<Void> from the protocol and just implement it in the extension.

Result:

More narrow protocol.

### CODE REVIEW

**Comment by normanmaurer** [comment]
As discussed before.... This needs a new major version so we should not merge it yet.

**Comment by weissi** [comment]
@normanmaurer we could do this one now :)

**Comment by normanmaurer** [comment]
Yes... let me do a pr

File: Sources/NIO/ChannelInvoker.swift [hunk 1]
```diff
@@ -132,47 +84,82 @@ public protocol ChannelOutboundInvoker {
...
+/// Extra `ChannelOutboundInvoker` methods. Each method that returns a `EventLoopFuture` will just do the following:
 ///     - create a new `EventLoopPromise<Void>`
 ///     - call the corresponding method that takes a `EventLoopPromise<Void>`
 ///     - return `EventLoopPromise.futureResult`
 extension ChannelOutboundInvoker {
+
+    /// Register on an `EventLoop` and so have all its IO handled.
+    ///
+    /// - returns: the future which will be notified once the operation completes.
     public func register() -> EventLoopFuture<Void> {
```
Comment (weissi) [inline Sources/NIO/ChannelInvoker.swift]: all of these should to take `file: StaticString = #file, line: UInt = #line` and pass them into the `newPromise` function so the 'leaked promise detection' will work with these guys

**Comment by IOOI-SqAR** [comment]
is there a reason why this isn't scheduled for the 2.0.0 Milestone?

**Comment by weissi** [comment]
@IOOI-SqAR thanks, it now is

@normanmaurer ping

**Comment by normanmaurer** [comment]
@weissi done... PTAL again

**Pull request merged**